src/mprismetadata.cpp

```diff
@@ -417,20 +426,22 @@ MprisMetaData::MprisMetaData(QObject *parent)
 
 QVariant MprisMetaData::trackId() const
 {
+    // Returns trackId in string format although for D-Bus we want it to look like an object path
     if (priv->m_metaData.contains(MetaFieldTrackId)) {
-        QString trackId = priv->m_metaData[MetaFieldTrackId].toString();
+        QString trackId = priv->m_metaData[MetaFieldTrackId].value<QDBusObjectPath>().path();
         if (trackId.startsWith(TrackObjectPathPrefix)) {
```

Comment:
I think this bit should be removed while you're here. I don't think we should be prepending anything really. It's up to the player to come up with a sane trackID

Reply:
I was going to wonder if that might break something but this code was never used because the line before was just wrong. Perhaps removing it would be the best thing to do.

Reply:
Yeah although it's a separate issue, it's about the only piece of this method not touched so it seems silly to preserve it. And you're right, it's never done anything, but could cause problems if it decides to. Probably best to remove the inverse operation at the same time though.

Reply:
> I was going to wonder if that might break something but this code was never used because the line before was just wrong. Perhaps removing it would be the best thing to do.

Except... it would have worked if the value stored was `QString` which it likely had been since `QDBusObjectPath` would not have survived to be delivered over D-Bus (conversion happens in `typedMetaData()` which I missed apparently). I think I need to revert my thinking a bit and maybe go back to converting the values in `MprisClient` when fetching them, otherwise I break those ways that this actually could have worked at all.

I should probably first take a small break though to clear my thoughts so I won't spend time uselessly going back and forth any more. Sorry for all the hassle.

Reply:
I'm not going to remove that now after all.